### ISSUE TITLE
Unable to debug exiv2 application code from Xcode

### ISSUE THREAD

**Issue by clanmills** [Exiv2/exiv2#374]
I can't debug into the application code in the exiv2 application from Xcode.  This was working in v0.26.

Application code (such as actions.cpp, exiv2.cpp) are shown in the debugger as 'pseudo machine code'.  The debugging is stepping machine code instead of src/actions.cpp and src/exiv2.cpp.

I've tried setting the CMake debug flag and CXXFLAGS without success
```
cmake -DCMAKE_BUILD_TYPE=Debug .. -G Xcode
env CXXFLAGS=-g cmake .. -G Xcode
```
The build output reports:
```
-- Debug:             -g
-- Release:           -O3 -DNDEBUG
-- RelWithDebInfo:    -O2 -g -DNDEBUG
-- MinSizeRel:        -Os -DNDEBUG
```

I think the default is 'Debug -g' and that applies to the library.  However it doesn't seem to apply to samples and applications.

**Comment by clanmills** [comment]
I have a fix:
```
733 rmills@rmillsmm:~/gnu/github/exiv2/exiv2/build $ git diff
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 91469b66..a05c537c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -306,6 +306,7 @@ if(EXIV2_BUILD_EXIV2_COMMAND)
     add_executable( exiv2 ${EXIV2_SRC} )
     set_target_properties( exiv2 PROPERTIES
         COMPILE_FLAGS ${EXTRA_COMPILE_FLAGS}
+        XCODE_ATTRIBUTE_GCC_GENERATE_DEBUGGING_SYMBOLS[variant=Debug] "YES"
     )
 
     target_compile_definitions(exiv2 PRIVATE EXV_LOCALEDIR="${CMAKE_INSTALL_LOCALEDIR}" )
734 rmills@rmillsmm:~/gnu/github/exiv2/exiv2/build $ 
```

Remarkably, we don't appear to require this for samples.

**Comment by piponazo** [comment]
> -- Debug:             -g
-- Release:           -O3 -DNDEBUG
-- RelWithDebInfo:    -O2 -g -DNDEBUG
-- MinSizeRel:        -Os -DNDEBUG

It's interesting to see that you do not have "-O0" for the Debug configuration there. In the file `cmake/compilerFlags.cmake` we have this line at the very beginning:

```
    set (CMAKE_CXX_FLAGS_DEBUG      "-g3 -gstrict-dwarf -O0")
```

**Comment by clanmills** [comment]
I've done a `$ git pull` and built.  You're right.  CMake reports:

```
-- Debug:             -g3 -gstrict-dwarf -O0
-- Release:           -O3 -DNDEBUG
-- RelWithDebInfo:    -O2 -g -DNDEBUG
-- MinSizeRel:        -Os -DNDEBUG
```

I still need my patch above to debug exiv2(.exe).   exifprint(.exe) can also be debugged.  I've submitted a PR #393  This might even be the first time I have successfully submitted a PR.  I didn't use the clanmills/exiv2 trampoline.

**Comment by clanmills** [comment]
Thanks @piponazo for your help with this.  As we've closed #388, I'm also closing this.
